packages/lesswrong/components/common/KarmaDisplay.tsx

```diff
@@ -2,10 +2,16 @@ import React from "react";
 import { registerComponent, Components } from "../../lib/vulcan-lib";
 import type { PopperPlacementType } from "@material-ui/core/Popper";
 import { forumTypeSetting } from "../../lib/instanceSettings";
+import { AnnualReviewMarketInfo, highlightMarket } from "../../lib/annualReviewMarkets";
+import { Link } from "../../lib/reactRouterWrapper";
+import { commentGetPageUrl } from "../../lib/collections/comments/helpers";
+import { useSingle } from "../../lib/crud/withSingle";
 
-const KarmaDisplay = ({document, placement="left"}: {
+const KarmaDisplay = ({document, placement="left", annualReviewMarketInfo = null, annualReviewMarketCommentId = null}: {
   document: VoteableType,
   placement?: PopperPlacementType,
+  annualReviewMarketInfo? : AnnualReviewMarketInfo | null,
+  annualReviewMarketCommentId? : string | null,
```

Comment:
if these are optional (`?:`) they don't also need to be `| null`

Reply:
I think that only accepts `undefined`

Reply:
https://www.typescriptlang.org/play?#code/C4TwDgpgBAZg9nKBeKBvAUFLUCGB+KALigGdgAnASwDsBzdAX3XQGM5qyoAjHc5KABQwisBAEpiZKnWQA+NJmzkIwAK7lqsPADocUPAQDkhxsx7kBqHMWqqANnYZigA